python/hsfs/core/online_store_sql_client.py

```diff
@@ -668,7 +669,9 @@ def serving_keys(self) -> Set[ServingKey]:
                 "Build serving keys from prepared statements ignoring prefix to ensure compatibility with older version."
             )
             self._serving_keys = util.build_serving_keys_from_prepared_statements(
-                self.prepared_statements[self.SINGLE_VECTOR_KEY],
+                self.prepared_statements[
+                    self.BATCH_VECTOR_KEY
```

Comment:
not clear why we changed the key from SINGLE to BATCH here?

Reply:
When there is a label fg, batch vector retrieval uses the fg_label prepared statement where as for single vector it does not. So when building keys/prefix etc... we miss the label prepared statement if we use the single vector. As to exactly the current implementation does that, I just don't exactly know